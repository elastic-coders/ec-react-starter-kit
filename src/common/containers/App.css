$blue: #056ef0;

.title {
  color: $blue;
}
