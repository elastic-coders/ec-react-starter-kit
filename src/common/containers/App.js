import React, { PureComponent, PropTypes as T } from 'react';
import { Link } from 'react-router'
import { connect } from 'react-redux';
import { push } from 'react-router-redux';

import styles from './App.css';

import { fetchThings } from '../store/actions';

@connect(state => ({
  things: state.test.thingsFetched,
}), {
  onFetchThings: fetchThings,
  changePage: push,
})
export default class App extends PureComponent {
  static propTypes = {
    things: T.string,
    onFetchThings: T.func,
    changePage: T.func,
  };

  render() {
    let children = null;
    if (this.props.children) {
      children = React.cloneElement(this.props.children, {
        auth: this.props.route.auth // sends auth instance from route to children
      })
    }

    return (
      <div>
        <h1 className={styles.title}>Hello, world!</h1>
        <Link to="/login">login</Link>
        <Link to="/private">private</Link>
        <button onClick={() => this.props.onFetchThings('yes!').then(() => console.log('yep it\'s a promise'))}>
          Fetched things: {this.props.things}
        </button>
        {this.props.route && this.props.route.auth.loggedIn() && (
          <button onClick={this.logout.bind(this)}>Logout</button>
        )}
        {children}
      </div>
    );
  }

  logout() {
    // destroys the session data
    this.props.route.auth.logout();
    // redirects to login page
    this.props.changePage('/login');
  }
}
