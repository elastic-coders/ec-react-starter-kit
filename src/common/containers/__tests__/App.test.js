import React from 'react';
import App from '../App';
import renderer from 'react-test-renderer';

it('renders things', () => {
  const component = renderer.create(
    <App things="something" />
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
