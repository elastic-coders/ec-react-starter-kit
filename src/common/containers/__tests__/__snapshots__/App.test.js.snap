exports[`test renders things 1`] = `
<div>
  <h1
    className="title">
    Hello, world!
  </h1>
  <a
    onClick={[Function bound handleClick]}
    style={Object {}}>
    login
  </a>
  <a
    onClick={[Function bound handleClick]}
    style={Object {}}>
    private
  </a>
  <button
    onClick={[Function onClick]}>
    Fetched things: 
    something
  </button>
</div>
`;
